README.md
# My [Home Manager] configuration

> [Nix] and [Home Manager] powered userspace applications

These are my personal dotfiles and userspace applications, done the Nix way.
They focus on providing a baseline, reproducible environment for development and productivity.

This is still very much a WIP, but I guess repos like this are never quite done. Still, it's been my daily driver for a while and it feels solid enough.

## Features
- [x] Select [Nerd Fonts] for patched fonts with extra glyphs
- [x] Productivity and QoL tools such as htop, mcfly, tldr, ripgrep and zoxide
- [x] Text editors: vim, neovim, kakoune and helix <sub>Why, yes, I *have* been experimenting with modal editors, how could you tell?</sub>
- [x] ZSH with oh-my-zsh plugins
- [x] [Nushell](https://www.nushell.sh/)
- [x] Starship prompt

## How to use
- [Download a release](https://github.com/lpchaim/home-manager/releases/latest)
- Extract to `~/.config/home-manager`
- Import `customizations.nix` into your `home.nix`
  - I.e. add `imports = [ ./customizations.nix ];`
  - Disabling modules is as easy as commenting the corresponding line out
- Run `home-manager switch`

[ansible]: https://www.ansible.com/
[home manager]: https://nix-community.github.io/home-manager/
[nerd fonts]: https://www.nerdfonts.com/
[nix]: https://nixos.org/


CHANGELOG.md
# Changelog

## [2.0.0](https://github.com/lpchaim/home-manager/compare/v1.1.0...v2.0.0) (2023-12-02)


### ⚠ BREAKING CHANGES

* Change root namespace
* Change namespace

### Features

* Add bash ([7c13094](https://github.com/lpchaim/home-manager/commit/7c13094b31fa8ed89a6d1c08f1f9417c2b788004))
* Add GUI applications ([fb00da4](https://github.com/lpchaim/home-manager/commit/fb00da4d0240a7a1a08141d4fddfbb2b21f0b6e8))
* Add laptop home configuration ([6f93aaf](https://github.com/lpchaim/home-manager/commit/6f93aafda23742a6eb7c40919e9ea7ead28fd4a0))
* Add ncdu ([5a9ae67](https://github.com/lpchaim/home-manager/commit/5a9ae67980c98aa151922f64d51cc5e3d35e3f0f))
* Add neofetch ([36f48d1](https://github.com/lpchaim/home-manager/commit/36f48d12119c70f0fad356df69f1e0583c1d5ded))
* Add proper module options ([5ca2dab](https://github.com/lpchaim/home-manager/commit/5ca2dab443d8dad6c2f48061c8f4a9243bf397f3))
* Change namespace ([45a36ff](https://github.com/lpchaim/home-manager/commit/45a36ff7113f5618fd6f1d885de60e9d00cc8c6c))
* **de:** Add desktop environment tweaks ([c29e7bd](https://github.com/lpchaim/home-manager/commit/c29e7bdb187e1762ac9114fdd8cbdba26e149387))
* **essentials:** Add clipboard utilities ([3ec692d](https://github.com/lpchaim/home-manager/commit/3ec692db57bd40d424235a5b893a8a9a521b163b))
* **essentials:** Add delta and difftastic ([dee5d2c](https://github.com/lpchaim/home-manager/commit/dee5d2c2ab277e9dfe58b57c7e34e915bb781eab))
* **helix:** Add file picker in current dir shortcut ([dca1562](https://github.com/lpchaim/home-manager/commit/dca1562fda9391c9e6a6adb05596d78ead6670c4))
* Improve GNOME theme ([b6f6a4e](https://github.com/lpchaim/home-manager/commit/b6f6a4e010a0ecfd6d9f2b88ccc4d822dd1e4cd7))
* **neovim:** Initial custom neovim configuration ([737326c](https://github.com/lpchaim/home-manager/commit/737326cb3cb83bd027d69fb79ffac46222c66f41))
* **starship:** Custom PHP icon ([66014d7](https://github.com/lpchaim/home-manager/commit/66014d785b5c4dcd146ce21c1f51347fcede143b))
* **tmux:** Add catppuccin theme ([95d1a50](https://github.com/lpchaim/home-manager/commit/95d1a501bbc30f3f05921dd49cffb2472dc58b75))
* **tmux:** Add proper theme option ([778c825](https://github.com/lpchaim/home-manager/commit/778c8256dd269c1d5cf721b5da34af3ce56f92c1))
* **tmux:** Change shortcut from A to Space ([ccef93e](https://github.com/lpchaim/home-manager/commit/ccef93e1802fc31b759f172a34de8a45c8cf4a1a))
* **tmux:** Enable auto renumber ([16ce0e9](https://github.com/lpchaim/home-manager/commit/16ce0e903d6d819e9457f9dab9cd97422a46e6e3))
* **tmux:** Improve shortcuts ([2fab9d6](https://github.com/lpchaim/home-manager/commit/2fab9d61d562b1849ffc3468944286d598bf7ead))
* **tmux:** Keep zoom state when navigating between panes ([7e300c4](https://github.com/lpchaim/home-manager/commit/7e300c44c427c5fb3350aee7dc3ecbcc0f7d471c))
* **tmux:** Resurrect commands with arguments ([608f3e1](https://github.com/lpchaim/home-manager/commit/608f3e18851202e5c28cc2b2784eb905140b0c53))
* **tmux:** Swap powerline for tmux-powerline ([d9aa4cf](https://github.com/lpchaim/home-manager/commit/d9aa4cff862187008d5133b761305d65b089d6fc))
* **tmux:** Tmux improvements ([857cf9c](https://github.com/lpchaim/home-manager/commit/857cf9c887de43f25ea4cd237f751be6c7ad4419))
* **tmux:** Tweak config ([c4aa473](https://github.com/lpchaim/home-manager/commit/c4aa4731935350c90ac21de57aeb1f9b5e8e4c9a))
* **tmux:** Tweak settings for proper color support ([aff96cf](https://github.com/lpchaim/home-manager/commit/aff96cfc074f76090a52fd77ab252446fcc5bb39))


### Bug Fixes

* Add Home Manager files back to .gitignore ([8392fc5](https://github.com/lpchaim/home-manager/commit/8392fc5391abd316c609744da4a5b4c4a63c2487))
* Base trait enables default modules ([921c3fa](https://github.com/lpchaim/home-manager/commit/921c3fa0f688d0be7e32cfea1622fffc885e4e94))
* Clipboard enhancements ([7e866c2](https://github.com/lpchaim/home-manager/commit/7e866c2320dedb4b6cdc3b02a5ad9a92d3e54590))
* **essentials:** Set TERM to screen-256color ([4c2a161](https://github.com/lpchaim/home-manager/commit/4c2a161796747bab6bafe622c3922bc67836daa7))
* **git:** Enable and improve git config ([aa487d9](https://github.com/lpchaim/home-manager/commit/aa487d9f8254317a9e07debec38169d2e0978ac1))
* **helix:** Remove auto-pairs ([4df5d78](https://github.com/lpchaim/home-manager/commit/4df5d781ea75ea8b769247bb7b31cc2077f34751))
* Pre-commit dependencies ([5a59398](https://github.com/lpchaim/home-manager/commit/5a59398cbecb125fab1167212242dac9dc66979c))
* **tmux-thumbs:** Custom shortcuts overriding thumbs activation key ([a1cfae9](https://github.com/lpchaim/home-manager/commit/a1cfae9a97a2b30cdafbb06588d5b8b8e13679a7))
* **tmux:** Copy commands ([10c6be7](https://github.com/lpchaim/home-manager/commit/10c6be7ae3bea8bb192661000706f6e4a19733e4))
* **tmux:** Fix resurrect and continuum ([e18e0e8](https://github.com/lpchaim/home-manager/commit/e18e0e8e2536b0ab97dbaf5e55ca65c3460c8d5f))
* **tmux:** Fix terminal type and colors ([58a9a4c](https://github.com/lpchaim/home-manager/commit/58a9a4c7784ddcb920c36f31da3336b9de01ad6c))


### Miscellaneous Chores

* Change root namespace ([9945a65](https://github.com/lpchaim/home-manager/commit/9945a65b296e7a81418d6e79bf042ca171d49803))

## [1.1.0](https://github.com/lpchaim/home-manager/compare/v1.0.0...v1.1.0) (2023-10-19)


### Features

* Add tmux ([5442697](https://github.com/lpchaim/home-manager/commit/54426979be35abb45b3f2a3ee0e51e9c707ecd12))

## 1.0.0 (2023-10-15)


### Features

* Add release-please action ([1d3731c](https://github.com/lpchaim/home-manager/commit/1d3731cf5de8b2d95af99ebdc6007d22d511a8a7))
* Create .gitignore ([0f6b552](https://github.com/lpchaim/home-manager/commit/0f6b5528c0f2c30a2541d63a3d97cbb86c09643c))
* initial configuration ([98f8d48](https://github.com/lpchaim/home-manager/commit/98f8d48f419d5e92c066556b6eda8dfa72afcb94))


### Bug Fixes


modules/tmux/tmux-powerline/config.example.sh
# Default configuration file for tmux-powerline.
# Modeline {
#	 vi: foldmarker={,} foldmethod=marker foldlevel=0 tabstop=4 filetype=sh
# }

# General {
	# Show which segment fails and its exit code.
	export TMUX_POWERLINE_DEBUG_MODE_ENABLED="false"
	# Use patched font symbols.
	export TMUX_POWERLINE_PATCHED_FONT_IN_USE="true"

	# The theme to use.
	export TMUX_POWERLINE_THEME="default"
	# Overlay directory to look for themes. There you can put your own themes outside the repo. Fallback will still be the "themes" directory in the repo.
	export TMUX_POWERLINE_DIR_USER_THEMES="${XDG_CONFIG_HOME:-$HOME/.config}/tmux-powerline/themes"
	# Overlay directory to look for segments. There you can put your own segments outside the repo. Fallback will still be the "segments" directory in the repo.
	export TMUX_POWERLINE_DIR_USER_SEGMENTS="${XDG_CONFIG_HOME:-$HOME/.config}/tmux-powerline/segments"

	# The initial visibility of the status bar. Can be {"on, off"}.
	export TMUX_POWERLINE_STATUS_VISIBILITY="on"
	# The status bar refresh interval in seconds.
	# Note that events that force-refresh the status bar (such as window renaming) will ignore this.
	export TMUX_POWERLINE_STATUS_INTERVAL="1"
	# The location of the window list. Can be {"absolute-centre, centre, left, right"}.
	# Note that "absolute-centre" is only supported on `tmux -V` >= 3.2.
	export TMUX_POWERLINE_STATUS_JUSTIFICATION="centre"

	# The maximum length of the left status bar.
	export TMUX_POWERLINE_STATUS_LEFT_LENGTH="60"
	# The maximum length of the right status bar.
	export TMUX_POWERLINE_STATUS_RIGHT_LENGTH="90"

	# Uncomment these if you want to enable tmux bindings for muting (hiding) one of the status bars.
	# E.g. this example binding would mute the left status bar when pressing <prefix> followed by Ctrl-[
	#export TMUX_POWERLINE_MUTE_LEFT_KEYBINDING="C-["
	#export TMUX_POWERLINE_MUTE_RIGHT_KEYBINDING="C-]"
# }

# battery.sh {
	# How to display battery remaining. Can be {percentage, cute}.
	export TMUX_POWERLINE_SEG_BATTERY_TYPE="percentage"
	# How may hearts to show if cute indicators are used.
	export TMUX_POWERLINE_SEG_BATTERY_NUM_HEARTS="5"
# }

# date.sh {
	# date(1) format for the date. If you don't, for some reason, like ISO 8601 format you might want to have "%D" or "%m/%d/%Y".
	export TMUX_POWERLINE_SEG_DATE_FORMAT="%F"
# }

# disk_usage.sh {
	# Filesystem to retrieve disk space information. Any from the filesystems available (run "df | awk '{print }'" to check them).
	export TMUX_POWERLINE_SEG_DISK_USAGE_FILESYSTEM="/"
# }

# earthquake.sh {
	# The data provider to use. Currently only "goo" is supported.
	export TMUX_POWERLINE_SEG_EARTHQUAKE_DATA_PROVIDER="goo"
	# How often to update the earthquake data in seconds.
	# Note: This is not an early warning detector, use this
	# to be informed about recent earthquake magnitudes in your
	# area. If this is too often, goo may decide to ban you form
	# their server
	export TMUX_POWERLINE_SEG_EARTHQUAKE_UPDATE_PERIOD="600"
	# Only display information when earthquakes are within this many minutes
	export TMUX_POWERLINE_SEG_EARTHQUAKE_ALERT_TIME_WINDOW="60"
	# Display time with this format
	export TMUX_POWERLINE_SEG_EARTHQUAKE_TIME_FORMAT='(%H:%M)'
	# Display only if magnitude is greater or equal to this number
	export TMUX_POWERLINE_SEG_EARTHQUAKE_MIN_MAGNITUDE="3"
# }

# hostname.sh {
	# Use short or long format for the hostname. Can be {"short, long"}.
	export TMUX_POWERLINE_SEG_HOSTNAME_FORMAT="short"
# }

# macos_notification_count.sh {
	# App ids to query in notification center, separated by space
	# To get the app id that is associated with a specific app run:
	# sqlite3 -list "/com.apple.notificationcenter/db/db" 'select * from app_info'
	# The first column contains the app ids
	# "5" is the app id of Messages.app
	# Only "banner" notifications are supported (see settings in the notification center)
	export TMUX_POWERLINE_SEG_MACOS_NOTIFICATION_COUNT_APPIDS="5"
	# Notification symbol
	export TMUX_POWERLINE_SEG_MACOS_NOTIFICATION_COUNT_CHAR="💬"
# }

# mailcount.sh {
	# Mailbox type to use. Can be any of {apple_mail, gmail, maildir, mbox, mailcheck}
	export TMUX_POWERLINE_SEG_MAILCOUNT_MAILBOX_TYPE=""
	
	## Gmail
	# Enter your Gmail username here WITH OUT @gmail.com.( OR @domain)
	export TMUX_POWERLINE_SEG_MAILCOUNT_GMAIL_USERNAME=""
	# Google password. Recomenned to use application specific password (https://accounts.google.com/b/0/IssuedAuthSubTokens) Leave this empty to get password from OS X keychain.
	# For OSX users : MAKE SURE that you add a key to the keychain in the format as follows
	# Keychain Item name : http://<value-you-fill-in-server-variable-below>
	# Account name : <username-below>@<server-below>
	# Password : Your password ( Once again, try to use 2 step-verification and application-specific password)
	# See http://support.google.com/accounts/bin/answer.py?hl=en&answer=185833 for more info.
	export TMUX_POWERLINE_SEG_MAILCOUNT_GMAIL_PASSWORD=""
	# Domain name that will complete your email. For normal GMail users it probably is "gmail.com but can be "foo.tld" for Google Apps users.
	export TMUX_POWERLINE_SEG_MAILCOUNT_GMAIL_SERVER="gmail.com"
	# How often in minutes to check for new mails.
	export TMUX_POWERLINE_SEG_MAILCOUNT_GMAIL_INTERVAL="5"
	
	## Maildir
	# Path to the maildir to check.
	export TMUX_POWERLINE_SEG_MAILCOUNT_MAILDIR_INBOX="/home/cheina/.mail/inbox/new"
	
	## mbox
	# Path to the mbox to check.
	export TMUX_POWERLINE_SEG_MAILCOUNT_MBOX_INBOX=""
	
	## mailcheck
	# Optional path to mailcheckrc
	export TMUX_POWERLINE_SEG_MAILCOUNT_MAILCHECKRC="/home/cheina/.mailcheckrc"
# }

# now_playing.sh {
	# Music player to use. Can be any of {audacious, banshee, cmus, apple_music, itunes, lastfm, mocp, mpd, mpd_simple, pithos, playerctl, rdio, rhythmbox, spotify, spotify_wine, file}.
	export TMUX_POWERLINE_SEG_NOW_PLAYING_MUSIC_PLAYER="spotify"
	# File to be read in case the song is being read from a file
	export TMUX_POWERLINE_SEG_NOW_PLAYING_FILE_NAME=""
	# Maximum output length.
	export TMUX_POWERLINE_SEG_NOW_PLAYING_MAX_LEN="40"
	# How to handle too long strings. Can be {trim, roll}.
	export TMUX_POWERLINE_SEG_NOW_PLAYING_TRIM_METHOD="trim"
	# Charcters per second to roll if rolling trim method is used.
	export TMUX_POWERLINE_SEG_NOW_PLAYING_ROLL_SPEED="2"
	
	# Hostname for MPD server in the format "[password@]host"
	export TMUX_POWERLINE_SEG_NOW_PLAYING_MPD_HOST="localhost"
	# Port the MPD server is running on.
	export TMUX_POWERLINE_SEG_NOW_PLAYING_MPD_PORT="6600"
	# Song display format for mpd_simple. See mpc(1) for delimiters.
	export TMUX_POWERLINE_SEG_NOW_PLAYING_MPD_SIMPLE_FORMAT="%artist% - %title%"
	# Song display format for playerctl. see "Format Strings" in playerctl(1).
	export TMUX_POWERLINE_SEG_NOW_PLAYING_PLAYERCTL_FORMAT="{{ artist }} - {{ title }}"
	# Song display format for rhythmbox. see "FORMATS" in rhythmbox-client(1).
	export TMUX_POWERLINE_SEG_NOW_PLAYING_RHYTHMBOX_FORMAT="%aa - %tt"
	
	# Last.fm
	# Set up steps for Last.fm
	# 1. Make sure jq(1) is installed on the system.
	# 2. Create a new API application at https://www.last.fm/api/account/create (name it tmux-powerline) and copy the API key and insert it below in the setting TMUX_POWERLINE_SEG_NOW_PLAYING_LASTFM_API_KEY
	# 3. Make sure the API can access your recently played song by going to you user privacy settings https://www.last.fm/settings/privacy and make sure "Hide recent listening information" is UNCHECKED.
	# Username for Last.fm if that music player is used.
	export TMUX_POWERLINE_SEG_NOW_PLAYING_LASTFM_USERNAME=""
	# API Key for the API.
	export TMUX_POWERLINE_SEG_NOW_PLAYING_LASTFM_API_KEY=""
	# How often in seconds to update the data from last.fm.
	export TMUX_POWERLINE_SEG_NOW_PLAYING_LASTFM_UPDATE_PERIOD="30"
	# Fancy char to display before now playing track
	export TMUX_POWERLINE_SEG_NOW_PLAYING_NOTE_CHAR="♫"
# }

# pwd.sh {
	# Maximum length of output.
	export TMUX_POWERLINE_SEG_PWD_MAX_LEN="40"
# }

# time.sh {
	# date(1) format for the time. Americans might want to have "%I:%M %p".
	export TMUX_POWERLINE_SEG_TIME_FORMAT="%H:%M"
# }

# tmux_session_info.sh {
	# Session info format to feed into the command: tmux display-message -p
	# For example, if FORMAT is '[ #S ]', the command is: tmux display-message -p '[ #S ]'
	export TMUX_POWERLINE_SEG_TMUX_SESSION_INFO_FORMAT="#S:#I.#P"
# }

# utc_time.sh {
	# date(1) format for the UTC time.
	export TMUX_POWERLINE_SEG_UTC_TIME_FORMAT="%H:%M %Z"
# }

# vcs_branch.sh {
	# Max length of the branch name.
	export TMUX_POWERLINE_SEG_VCS_BRANCH_MAX_LEN="24"
# }

# weather.sh {
	# The data provider to use. Currently only "yahoo" is supported.
	export TMUX_POWERLINE_SEG_WEATHER_DATA_PROVIDER="yrno"
	# What unit to use. Can be any of {c,f,k}.
	export TMUX_POWERLINE_SEG_WEATHER_UNIT="c"
	# How often to update the weather in seconds.
	export TMUX_POWERLINE_SEG_WEATHER_UPDATE_PERIOD="600"
	# Name of GNU grep binary if in PATH, or path to it.
	export TMUX_POWERLINE_SEG_WEATHER_GREP="grep"
	# Location of the JSON parser, jq
	export TMUX_POWERLINE_SEG_WEATHER_JSON="jq"
	# Your location
	# Latitude and Longtitude for use with yr.no
	TMUX_POWERLINE_SEG_WEATHER_LAT=""
	TMUX_POWERLINE_SEG_WEATHER_LON=""
# }


modules/tmux/tmux-powerline/config.sh
# Default configuration file for tmux-powerline.
# Modeline {
#	 vi: foldmarker={,} foldmethod=marker foldlevel=0 tabstop=4 filetype=sh
# }

# TMUX_POWERLINE_RIGHT_STATUS_SEGMENTS=""
# TMUX_POWERLINE_LEFT_STATUS_SEGMENTS=""

# General {
	# Show which segment fails and its exit code.
	export TMUX_POWERLINE_DEBUG_MODE_ENABLED="false"
	# Use patched font symbols.
	export TMUX_POWERLINE_PATCHED_FONT_IN_USE="true"

	# The theme to use.
	export TMUX_POWERLINE_THEME="custom"
	# Overlay directory to look for themes. There you can put your own themes outside the repo. Fallback will still be the "themes" directory in the repo.
	export TMUX_POWERLINE_DIR_USER_THEMES="${XDG_CONFIG_HOME:-$HOME/.config}/tmux-powerline/themes"
	# Overlay directory to look for segments. There you can put your own segments outside the repo. Fallback will still be the "segments" directory in the repo.
	export TMUX_POWERLINE_DIR_USER_SEGMENTS="${XDG_CONFIG_HOME:-$HOME/.config}/tmux-powerline/segments"

	# The initial visibility of the status bar. Can be {"on, off"}.
	export TMUX_POWERLINE_STATUS_VISIBILITY="on"
	# The status bar refresh interval in seconds.
	# Note that events that force-refresh the status bar (such as window renaming) will ignore this.
	export TMUX_POWERLINE_STATUS_INTERVAL="1"
	# The location of the window list. Can be {"absolute-centre, centre, left, right"}.
	# Note that "absolute-centre" is only supported on `tmux -V` >= 3.2.
	export TMUX_POWERLINE_STATUS_JUSTIFICATION="absolute-centre"

	# The maximum length of the left status bar.
	export TMUX_POWERLINE_STATUS_LEFT_LENGTH="60"
	# The maximum length of the right status bar.
	export TMUX_POWERLINE_STATUS_RIGHT_LENGTH="90"

	# Uncomment these if you want to enable tmux bindings for muting (hiding) one of the status bars.
	# E.g. this example binding would mute the left status bar when pressing <prefix> followed by Ctrl-[
	#export TMUX_POWERLINE_MUTE_LEFT_KEYBINDING="C-["
	#export TMUX_POWERLINE_MUTE_RIGHT_KEYBINDING="C-]"
# }

# battery.sh {
	# How to display battery remaining. Can be {percentage, cute}.
	export TMUX_POWERLINE_SEG_BATTERY_TYPE="percentage"
	# How may hearts to show if cute indicators are used.
	export TMUX_POWERLINE_SEG_BATTERY_NUM_HEARTS="5"
# }

# date.sh {
	# date(1) format for the date. If you don't, for some reason, like ISO 8601 format you might want to have "%D" or "%m/%d/%Y".
	export TMUX_POWERLINE_SEG_DATE_FORMAT="%F"
# }

# disk_usage.sh {
	# Filesystem to retrieve disk space information. Any from the filesystems available (run "df | awk '{print }'" to check them).
	export TMUX_POWERLINE_SEG_DISK_USAGE_FILESYSTEM="/"
# }

# earthquake.sh {
	# The data provider to use. Currently only "goo" is supported.
	export TMUX_POWERLINE_SEG_EARTHQUAKE_DATA_PROVIDER="goo"
	# How often to update the earthquake data in seconds.
	# Note: This is not an early warning detector, use this
	# to be informed about recent earthquake magnitudes in your
	# area. If this is too often, goo may decide to ban you form
	# their server
	export TMUX_POWERLINE_SEG_EARTHQUAKE_UPDATE_PERIOD="600"
	# Only display information when earthquakes are within this many minutes
	export TMUX_POWERLINE_SEG_EARTHQUAKE_ALERT_TIME_WINDOW="60"
	# Display time with this format
	export TMUX_POWERLINE_SEG_EARTHQUAKE_TIME_FORMAT='(%H:%M)'
	# Display only if magnitude is greater or equal to this number
	export TMUX_POWERLINE_SEG_EARTHQUAKE_MIN_MAGNITUDE="3"
# }

# hostname.sh {
	# Use short or long format for the hostname. Can be {"short, long"}.
	export TMUX_POWERLINE_SEG_HOSTNAME_FORMAT="short"
# }

# macos_notification_count.sh {
	# App ids to query in notification center, separated by space
	# To get the app id that is associated with a specific app run:
	# sqlite3 -list "/com.apple.notificationcenter/db/db" 'select * from app_info'
	# The first column contains the app ids
	# "5" is the app id of Messages.app
	# Only "banner" notifications are supported (see settings in the notification center)
	export TMUX_POWERLINE_SEG_MACOS_NOTIFICATION_COUNT_APPIDS="5"
	# Notification symbol
	export TMUX_POWERLINE_SEG_MACOS_NOTIFICATION_COUNT_CHAR="💬"
# }

# mailcount.sh {
	# Mailbox type to use. Can be any of {apple_mail, gmail, maildir, mbox, mailcheck}
	export TMUX_POWERLINE_SEG_MAILCOUNT_MAILBOX_TYPE=""
	
	## Gmail
	# Enter your Gmail username here WITH OUT @gmail.com.( OR @domain)
	export TMUX_POWERLINE_SEG_MAILCOUNT_GMAIL_USERNAME=""
	# Google password. Recomenned to use application specific password (https://accounts.google.com/b/0/IssuedAuthSubTokens) Leave this empty to get password from OS X keychain.
	# For OSX users : MAKE SURE that you add a key to the keychain in the format as follows
	# Keychain Item name : http://<value-you-fill-in-server-variable-below>
	# Account name : <username-below>@<server-below>
	# Password : Your password ( Once again, try to use 2 step-verification and application-specific password)
	# See http://support.google.com/accounts/bin/answer.py?hl=en&answer=185833 for more info.
	export TMUX_POWERLINE_SEG_MAILCOUNT_GMAIL_PASSWORD=""
	# Domain name that will complete your email. For normal GMail users it probably is "gmail.com but can be "foo.tld" for Google Apps users.
	export TMUX_POWERLINE_SEG_MAILCOUNT_GMAIL_SERVER="gmail.com"
	# How often in minutes to check for new mails.
	export TMUX_POWERLINE_SEG_MAILCOUNT_GMAIL_INTERVAL="5"
	
	## Maildir
	# Path to the maildir to check.
	export TMUX_POWERLINE_SEG_MAILCOUNT_MAILDIR_INBOX="$HOME/.mail/inbox/new"
	
	## mbox
	# Path to the mbox to check.
	export TMUX_POWERLINE_SEG_MAILCOUNT_MBOX_INBOX=""
	
	## mailcheck
	# Optional path to mailcheckrc
	export TMUX_POWERLINE_SEG_MAILCOUNT_MAILCHECKRC="$HOME/.mailcheckrc"
# }

# now_playing.sh {
	# Music player to use. Can be any of {audacious, banshee, cmus, apple_music, itunes, lastfm, mocp, mpd, mpd_simple, pithos, playerctl, rdio, rhythmbox, spotify, spotify_wine, file}.
	export TMUX_POWERLINE_SEG_NOW_PLAYING_MUSIC_PLAYER="spotify"
	# File to be read in case the song is being read from a file
	export TMUX_POWERLINE_SEG_NOW_PLAYING_FILE_NAME=""
	# Maximum output length.
	export TMUX_POWERLINE_SEG_NOW_PLAYING_MAX_LEN="40"
	# How to handle too long strings. Can be {trim, roll}.
	export TMUX_POWERLINE_SEG_NOW_PLAYING_TRIM_METHOD="trim"
	# Charcters per second to roll if rolling trim method is used.
	export TMUX_POWERLINE_SEG_NOW_PLAYING_ROLL_SPEED="2"
	
	# Hostname for MPD server in the format "[password@]host"
	export TMUX_POWERLINE_SEG_NOW_PLAYING_MPD_HOST="localhost"
	# Port the MPD server is running on.
	export TMUX_POWERLINE_SEG_NOW_PLAYING_MPD_PORT="6600"
	# Song display format for mpd_simple. See mpc(1) for delimiters.
	export TMUX_POWERLINE_SEG_NOW_PLAYING_MPD_SIMPLE_FORMAT="%artist% - %title%"
	# Song display format for playerctl. see "Format Strings" in playerctl(1).
	export TMUX_POWERLINE_SEG_NOW_PLAYING_PLAYERCTL_FORMAT="{{ artist }} - {{ title }}"
	# Song display format for rhythmbox. see "FORMATS" in rhythmbox-client(1).
	export TMUX_POWERLINE_SEG_NOW_PLAYING_RHYTHMBOX_FORMAT="%aa - %tt"
	
	# Last.fm
	# Set up steps for Last.fm
	# 1. Make sure jq(1) is installed on the system.
	# 2. Create a new API application at https://www.last.fm/api/account/create (name it tmux-powerline) and copy the API key and insert it below in the setting TMUX_POWERLINE_SEG_NOW_PLAYING_LASTFM_API_KEY
	# 3. Make sure the API can access your recently played song by going to you user privacy settings https://www.last.fm/settings/privacy and make sure "Hide recent listening information" is UNCHECKED.
	# Username for Last.fm if that music player is used.
	export TMUX_POWERLINE_SEG_NOW_PLAYING_LASTFM_USERNAME=""
	# API Key for the API.
	export TMUX_POWERLINE_SEG_NOW_PLAYING_LASTFM_API_KEY=""
	# How often in seconds to update the data from last.fm.
	export TMUX_POWERLINE_SEG_NOW_PLAYING_LASTFM_UPDATE_PERIOD="30"
	# Fancy char to display before now playing track
	export TMUX_POWERLINE_SEG_NOW_PLAYING_NOTE_CHAR="♫"
# }

# pwd.sh {
	# Maximum length of output.
	export TMUX_POWERLINE_SEG_PWD_MAX_LEN="40"
# }

# time.sh {
	# date(1) format for the time. Americans might want to have "%I:%M %p".
	export TMUX_POWERLINE_SEG_TIME_FORMAT="%H:%M"
# }

# tmux_session_info.sh {
	# Session info format to feed into the command: tmux display-message -p
	# For example, if FORMAT is '[ #S ]', the command is: tmux display-message -p '[ #S ]'
	export TMUX_POWERLINE_SEG_TMUX_SESSION_INFO_FORMAT="#S:#I.#P"
# }

# utc_time.sh {
	# date(1) format for the UTC time.
	export TMUX_POWERLINE_SEG_UTC_TIME_FORMAT="%H:%M %Z"
# }

# vcs_branch.sh {
	# Max length of the branch name.
	export TMUX_POWERLINE_SEG_VCS_BRANCH_MAX_LEN="24"
# }

# weather.sh {
	# The data provider to use. Currently only "yahoo" is supported.
	export TMUX_POWERLINE_SEG_WEATHER_DATA_PROVIDER="yrno"
	# What unit to use. Can be any of {c,f,k}.
	export TMUX_POWERLINE_SEG_WEATHER_UNIT="c"
	# How often to update the weather in seconds.
	export TMUX_POWERLINE_SEG_WEATHER_UPDATE_PERIOD="600"
	# Name of GNU grep binary if in PATH, or path to it.
	export TMUX_POWERLINE_SEG_WEATHER_GREP="grep"
	# Location of the JSON parser, jq
	export TMUX_POWERLINE_SEG_WEATHER_JSON="jq"
	# Your location
	# Latitude and Longtitude for use with yr.no
	TMUX_POWERLINE_SEG_WEATHER_LAT=""
	TMUX_POWERLINE_SEG_WEATHER_LON=""
# }
